### PR TITLE
Organizer dashboard

### PR DESCRIPTION
**Pull request by optimusinhisprime**
Items remaining:
- Making each event open its single event page (The single event page should allow for editing and deleting an event as an organizer. Adding a new stall too.)
- Adding a new event.

Screenshot (334)
Screenshot (335)
Screenshot (336)

### CODE REVIEW

**Comment by ChinyangaTL** [comment]
this looks cool

File: handlers/stallRequestHandler.js [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
+        path: "event",
+        select: "name eventStalls",
+        model: Event,
+      })
+      .populate({
+        path: "vendor",
+        select: "firstname surname extendProfile",
+        model: User,
+      })
+      .exec()
```
Comment (ChinyangaTL) [inline handlers/stallRequestHandler.js:21]: 👏

File: pages/organizer/dashboard.jsx [hunk 2]
```diff
@@ -1,24 +1,75 @@
...
+const OrganizerDashboard = ({session, events, stallRequests}) => {
+  return <Flex  direction="column" align="center">
     <Heading mt={10} as='h3' size='lg'>
     My Events
   </Heading>
   <Flex align="center" w="50%" direction="column" mt={8}>
-    <Event/>
+    {events.events.map((event)=>{
+      return <Event key={event._id} name={event.name} date={event.date} venue={event.venue} imageUrl={event.images[0]}/>
+    })}
```
Comment (ChinyangaTL) [inline pages/organizer/dashboard.jsx:16]: Why pass down the specific properties instead of everything?

File: pages/organizer/dashboard.jsx [hunk 3]
```diff
@@ -1,24 +1,75 @@
...
-  <StallRequest/>
+  <Flex mb={10} w="50%" direction="column" mt= {8}>
+  {stallRequests.stalls.map((request)=>{
+    const stallCategory = request.event.eventStalls.filter((stall)=>{ 
+      return stall.stallId == request.stallId;})
+      const userProfile = <VStack align="left">
+        <Text>Business Name: {request.vendor.extendProfile.businessName}</Text>
+        <Text>Store Number: {request.vendor.extendProfile.businessNumber}</Text>
+        <Text>Store Address: {request.vendor.extendProfile.businessAddress}</Text>
+        <Text>Store Bio: {request.vendor.extendProfile.businessDescription}</Text>
```
Comment (ChinyangaTL) [inline pages/organizer/dashboard.jsx:31]: These properties can be extracted from 'request.vendor.extendProfile' to make it cleaner
somethig like const {businessName, businessNumber} = request.vendor.extendProfile

File: pages/organizer/dashboard.jsx [hunk 4]
```diff
@@ -1,24 +1,75 @@
...
+    return {
+      redirect: {
+        destination: "/api/auth/signin",
+        permanent: false,
+      },
+    };
+  } else{
+    const eventsEndpoint = `${process.env.NEXTAUTH_URL}/api/v1/events?organizerId=${session.userId}`;
+    const eventsResponse = await fetch(eventsEndpoint);
+    events = await eventsResponse.json();
```
Comment (ChinyangaTL) [inline pages/organizer/dashboard.jsx:61]: Probably should have mentioned, but i introduces axios into the project

**Review by ChinyangaTL — APPROVED** [review]

File: pages/organizer/dashboard.jsx (hunk 4 above)
Comment (optimusinhisprime) [reply pages/organizer/dashboard.jsx:61]: Alright. I dont have it in the packages this yet

File: handlers/stallRequestHandler.js (hunk 1 above)
Comment (optimusinhisprime) [reply handlers/stallRequestHandler.js:21]: Danko

File: pages/organizer/dashboard.jsx (hunk 2 above)
Comment (optimusinhisprime) [reply pages/organizer/dashboard.jsx:16]: And then destructuring what i need in the event component. Ill clean it up

**Comment by optimusinhisprime** [comment]
@ChinyangaTL All good?

**Comment by ChinyangaTL** [comment]
@optimusinhisprime, yes sir

**Pull request merged**